docs/userguide/serialization.md

```diff
@@ -0,0 +1,390 @@
+
+```{admonition} Equivalent copies
+:class: important
+Roundtrip serialization is configured such that the obtained copy of an object
+is semantically equivalent to its original version and thus behaves identically.
+Note, however, that some objects contain non-persistent content (i.e., internal objects
+such as temporary data or cached computation results) that may be lost during a 
+serialization roundtrip.
+Because this content will be automatically recreated on the fly when needed,
+it is ignored for comparison with the `==` operator, enabling a semantically correct
+equality check.
+```
+
+For example:
+```python
+from baybe.parameters import CategoricalParameter
+
+parameter = CategoricalParameter(name="Setting", values=["low", "high"])
+json_string = parameter.to_json()
+reconstructed = CategoricalParameter.from_json(json_string)
+assert parameter == reconstructed
+```
+
+This form of roundtrip serialization can be used, for instance, to persist objects
+for long-term storage, but it also provides an easy way to "move" existing objects
+between Python sessions by executing the deserializing step in a different context
+than the serialization step.
+
+## Deserialization from configuration strings
+The workflow described [above](#JSON_SERIALIZATION) most naturally applies to
+situations where we start inside the Python ecosystem and want to make an object
+"leave" the running session. 
+However, in many cases, we would like to kickstart the process from the other end and
+rather specify a BayBE object **outside** Python for use in a later computation.
+Common examples are when we wish to interact with an API or simply want to persist 
+a certain BayBE component in the form of a "configuration" file.
+
+The following sections give an overview of the flexibilities that are offered for this
+task. Of course, the underlying concepts can be mixed and matched arbitrarily.
+
+### Basic string assembly
+Writing a configuration for a certain BayBE object in form of a serialization string is
+easy:
+1. Select your desired object class
+2. Identify the arguments expected by one of its constructors (see also [here](#ALTERNATIVE_CONSTRUCTORS))
+3. Pack them into a JSON string that mirrors the constructor signature
+
+Let's have a more detailed look, for instance, at the serialization string from
+the [above example](#JSON_SERIALIZATION), this time assuming we wanted to assemble
+the string manually.
+For this purpose, we have a peek at the signature of the `__init__` method of
+{meth}`CategoricalParameter <baybe.parameters.categorical.CategoricalParameter.__init__>`
+and notice that it has two required arguments, `name` and `values`.
+We specify these accordingly as separate fields in the JSON string:
+
+```python
+from baybe.parameters import CategoricalParameter
+
+parameter_str = """
+{
+    "name": "Setting",
+    "values": ["low", "high"]
+}
+"""
+from_json = CategoricalParameter.from_json(parameter_str)
```

Comment:
I suggest to rename all occurrences of variable names from `*from_json` to `*via_json` to i) avoid confusion with the method name and ii) be consistent with the other comparison variable often called `via_init`

Reply:
done